### ISSUE TITLE
Add functionality to vigilance.py to metric actual code coverage

### ISSUE THREAD

**Issue by bocajnotnef** [devosoft/Empirical#82]
So, once #76 gets merged this is something I'd like to work on.

So, the reason I made vigilance was that gcov wasn't properly detecting executable lines in the code and wasn't generating correct figures for diff-cover to use when checking to see if people test the code they add. I solved this problem by writing a diff-cover script that correctly interprets the broken gcov output and generates a helpful diff-cover.

My proposal is that I use the existing groundwork in vigilance and develop a way to easily display the test coverage of a file [a la lcov/gcovr](https://github.com/devosoft/Empirical/issues/72#issuecomment-222806814) in pretty pretty html.

Vigilance can already extract executable lines from a source file (thanks to helpful g++ flags) and diff them against existing coverage data--It wouldn't be too much more of a step to use that information to build the full coverage map of a file instead of just the lines changed in the diff.

This would be useful for finding areas of low coverage in the library, which we could spew low-hanging-fruit issues for.

Even if I don't write up HTML output it would be easy to render to the console or a file.

tl;dr: I think I can rewrite gcovr in a way that's useful to us to accurately metric coverage throughout Empirical, and I'd like to start working on thhat once #76 gets merged.

**Comment by bocajnotnef** [comment]
deprioritizing metricing test coverage
